shell/platform/linux/fl_engine.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_engine.h"
+#include "flutter/shell/platform/linux/fl_engine_private.h"
+#include "flutter/shell/platform/linux/fl_renderer.h"
+
+#include <gmodule.h>
+
+struct _FlEngine {
+  GObject parent_instance;
+
+  FlDartProject *project;
+  FlRenderer* renderer;
+  FLUTTER_API_SYMBOL(FlutterEngine) engine;
+};
+
+G_DEFINE_TYPE(FlEngine, fl_engine, G_TYPE_OBJECT)
+
+/* Callback from Flutter engine that are passed to the renderer */
+
+static void* fl_engine_gl_proc_resolver(void* user_data, const char* name) {
+  FlEngine* self = static_cast<FlEngine*>(user_data);
+  return fl_renderer_get_proc_address(self->renderer, name);
```

Comment:
These will all need to handle the case where `self->renderer` is null for headless mode, but we can add that later if you prefer.

Reply:
Yeah, I think it's best to do that later. I wasn't sure what the side-effects would be of just stubbing out all these callbacks.